# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1263433531 18000
#      Wed Jan 13 20:45:31 2010 -0500
# Node ID 29959c705df2a6d65f7945beaf246d6d1931b6fd
# Parent  d3451c1496f846fad0f520cabba0bd0bb2f29bdf
avoid assignment in condition of if statement

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-01-13  John W. Eaton  <jwe@octave.org>
+
+	* oct-obj.cc (octave_value_list::any_cell): Avoid assignment in
+	condition of if statement.
+
 2010-01-13  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/lu.cc: Avoid test failures if HAVE_QRUPDATE_LUU is
 	not defined.
 
 2010-01-12  David Bateman  <dbateman@free.fr>
 
 	* graphics.h.in (patch::properties): Add "flat" value to
diff --git a/src/oct-obj.cc b/src/oct-obj.cc
--- a/src/oct-obj.cc
+++ b/src/oct-obj.cc
@@ -185,24 +185,23 @@ octave_value_list::all_scalars (void) co
   for (i = 0; i < n && elem (i).is_string (); i++) ;
   
   return i == n;
 }
 
 bool
 octave_value_list::any_cell (void) const
 {
-  bool retval = false;
   octave_idx_type n = length ();
 
   for (octave_idx_type i = 0; i < n; i++)
-    if (retval = elem (i).is_cell ())
-      break;
-  
-  return retval;
+    if (elem (i).is_cell ())
+      return true;
+
+  return false;
 }
 
 bool
 octave_value_list::has_magic_colon (void) const
 {
   octave_idx_type n = length ();
 
   for (octave_idx_type i = 0; i < n; i++)
